internal/dataplane/kong_client.go

```diff
@@ -423,21 +424,29 @@ func (c *KongClient) Update(ctx context.Context) error {
 // enables filtering for which objects are currently applied to the data-plane,
 // as well as updating the c.kubernetesObjectStatusQueue to queue those objects
 // for reconciliation so their statuses can be properly updated.
-func (c *KongClient) triggerKubernetesObjectReport(objs ...client.Object) {
+func (c *KongClient) triggerKubernetesObjectReport(succeededObjs []client.Object, parsingErrors []parser.ParsingError) {
 	// first a new set of the included objects for the most recent configuration
 	// needs to be generated.
 	set := k8sobj.Set{}
-	for _, obj := range objs {
-		set.Insert(obj)
+	for _, obj := range succeededObjs {
+		set.InsertSucceeded(obj)
 	}
 
 	c.updateKubernetesObjectReportFilter(set)
 
+	failedObjs := make([]client.Object, 0, len(parsingErrors))
+	for _, err := range parsingErrors {
+		failedObjs = append(failedObjs, err.RelatedObjects()...)
+		set.InsertFailed(err.RelatedObjects()...)
+	}
+
 	// after the filter has been updated we signal the status queue so that the
 	// control-plane can update the Kubernetes object statuses for affected objs.
 	// this has to be done in a separate loop so that the filter is in place
 	// before the objects are enqueued, as the filter is used by the control-plane
+	objs := append(succeededObjs, failedObjs...)
 	for _, obj := range objs {
+		c.logger.Infof("publishing event for %s/%s/%s", obj.GetObjectKind(), obj.GetNamespace(), obj.GetName())
 		c.kubernetesObjectStatusQueue.Publish(obj)
 	}
 }
```

Comment:
Generic in-memory events are being published for both succeeded and failed objects.